Make selected text copyable instead of copying password (fix for #7059)

mckeema opened the pull request:
Fixes #7059. Checks whether the fingerprint, comment, or public key widgets of the SSH Agent tab in the Edit Entry screen are selected when a copy event occurs. If they are, the relevant selected text is copied, otherwise falls back to the current behavior of copying the password. The clipboard is cleared after the timeout if the usual setting is enabled. Below is a short video demonstration of the new behavior. Previously, none of the text mentioned above could be copied using the copy shortcut. The password would be copied instead.


## Screenshots
https://user-images.githubusercontent.com/40369019/144961908-12ed5f16-941d-4d75-9324-77c6ac7c96a4.mp4


## Testing strategy
I tested the change manually, as you can see in the video included above. I also ran `make test` locally and got the same results on my branch as the develop branch.


## Type of change
- ✅ Bug fix (non-breaking change that fixes an issue)

droidmonkey commented:
This is not the desired approach to this and is extremely fragile to name changes in the UI file. Additionally, all text input fields and labels suffer from this bug, even on the preview widget. Remove the checks for the widget names and add a QTextEdit* check as well.

mckeema committed:
Make selected text copyable instead of copying password

* Fixes 7209

mckeema commented:
Thanks for the pointers. No screenshots because the behavior hasn't changed from the original video. I removed the object name checks and replaced them with checks for QLabel, QTextEdit, and QPlainTextEdit. If any text is highlighted in any of those three widget types, it is copied instead of the password. Is this closer to the desired behavior/approach?

droidmonkey commented:
Yah that is perfect!
